eZ/Publish/Core/REST/Common/FieldTypeProcessor/RelationProcessor.php

```diff
@@ -37,7 +37,7 @@ public function postProcessFieldSettingsHash($outgoingSettingsHash)
         $outgoingSettingsHash = parent::postProcessFieldSettingsHash($outgoingSettingsHash);
 
         if (isset($outgoingSettingsHash['selectionRoot'])) {
-            $outgoingSettingsHash['selectionRootHref'] = $this->mapToContentHref(
+            $outgoingSettingsHash['selectionRootHref'] = $this->mapToLocationHref(
                 $outgoingSettingsHash['selectionRoot']
```

Comment:
We need to check what's stored as the selectionRoot value. If a location path is stored (1/2/42), then we're good. not good, and we need to load the matching location object, and get its actual path (1/2/42).

The easiest is to load a content item that has a relation field with this selectionRoot set, and see what comes out of this implementation. If it's a path like location/42, the we're not good:

`curl http://localhost:8000/api/ezp/v2/content/objects/<id> -H 'Accept: application/vnd.ez.api.Content+xml'`